aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -198,6 +243,19 @@ public AwsProperties(Map<String, String> properties) {
     this.s3FileIoAcl = ObjectCannedACL.fromValue(aclType);
     Preconditions.checkArgument(s3FileIoAcl == null || !s3FileIoAcl.equals(ObjectCannedACL.UNKNOWN_TO_SDK_VERSION),
         "Cannot support S3 CannedACL " + aclType);
+
+    this.glueCatalogLockEnabled = PropertyUtil.propertyAsBoolean(properties,
+        GLUE_CATALOG_LOCK_ENABLED, GLUE_CATALOG_LOCK_ENABLED_DEFAULT);
+    this.glueCatalogLockTable = properties.get(GLUE_CATALOG_LOCK_TABLE);
+    this.glueCatalogLockWaitMs = PropertyUtil.propertyAsLong(properties,
+        GLUE_CATALOG_LOCK_WAIT_MS, GLUE_CATALOG_LOCK_WAIT_MS_DEFAULT);
+    this.glueCatalogLockExpireMs = PropertyUtil.propertyAsLong(properties,
+        GLUE_CATALOG_LOCK_MS_EXPIRE, GLUE_CATALOG_LOCK_EXPIRE_MS_DEFAULT);
+
+    if (glueCatalogLockEnabled) {
+      Preconditions.checkNotNull(glueCatalogLockTable,
+          GLUE_CATALOG_LOCK_TABLE + " must not be null with GLue catalog lock is enabled");
```

Comment:
Typo: "GLue" -> "Glue"

Reply:
Also, I think we normally use "is required" instead of "must not be null". Null is what we get when it isn't set, but we don't need to call attention to that value. We need to tell the user what is wrong, which is that a required configuration is missing.